sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/StateFetchingIterators.java

```diff
@@ -91,6 +94,82 @@ public static <T> CachingStateIterable<T> readAllAndDecodeStartingFrom(
         valueCoder);
   }
 
+  public static <T> UncachedStateIterable<T> readAllAndDecodeStartingFrom(
+      BeamFnStateClient beamFnStateClient,
+      StateRequest stateRequestForFirstChunk,
+      Coder<T> valueCoder) {
+    return new UncachedStateIterable<>(beamFnStateClient, stateRequestForFirstChunk, valueCoder);
+  }
+
+  static class UncachedStateIterable<T> extends PrefetchableIterables.Default<T> {
```

Comment:
If this isn't used for state anywhere (I don't think it is) we probably don't even need to use the Prefethable* interfaces at all, simplifying things further.

Reply:
I meant to refer to the StateBackedIterable class itself. StateFetchingIterators are definitely used elsewhere.

Reply:
Sorry still not sure I'm understanding.  Are you suggesting just inlining this class in StateBackedIterable to avoid having it use the interfaces?

Reply:
I meant, e.g. we could probably change declaration at https://github.com/apache/beam/blob/release-2.60.0/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/StateBackedIterable.java#L77 to be just an Iterable, changing the constructor for StateBackedIterable[Coder] to not take a Cache at all, etc. But it's also not too much work to keep it as a Prefetchable iterable given the underlying iterable is.